### DISCUSSION TITLE
Ideas for optimization and standardization of socketify.py

### DISCUSSION THREAD

**Discussion by AmirHmZz** [cirospaciari/socketify.py#49] [Ideas]
As the first comment I want to make a checklist : 

- [ ] Move `Enums` and `Option` classes to other python files
- [x] Avoid global refs ( As we can see for `SocketRefs` in `socketify.py` ) which make access to variables slower a lot
- [x] Use `dataclass` for `Option` classes to reduce boilerplate code ( Merged in [this commit](https://github.com/cirospaciari/socketify.py/commit/6f90d2dde6a876f3766b060757041e3c7b197689) )
- [ ] Move handler validators out of trigger functions

**Comment by cirospaciari** [comment]
SocketRefs is needed to keep the ref alive in CFFI without this the ref count goes to 0 and GC cleans the socket data, with HPy this will be solved, we can also move it to the App instead of global just need to check if is faster or not, because each process can have only 1 App by design so is not really an problem.

@AmirHmZz Please let's use this discussion to talk about an major refactor also, we will create another branch to do it. We will test all after the refactor and also add unit tests.

I need to finish somethings before the refactor starts, but please send all the advice possible about Python. :heart: 

Also we can use the https://discord.socketify.dev/ i will create an channel for this.

**Reply by AmirHmZz** [reply]
Yes I know exactly why that is defined. We should talk about it.

**Reply by AmirHmZz** [reply]
The discord invite link is expired...

**Reply by cirospaciari** [reply]
> The discord invite link is expired...

https://discord.com/invite/sRUPx7F7XJ
Try again i set this to never expires and https://discord.socketify.dev is an auto redirect to an valid discord link

**Reply by AmirHmZz** [reply]
Again the same message appears : _This invite may be expired, or you might not have permission to join._
Maybe I don't have permission to access that link?

**Reply by cirospaciari** [reply]
What that crazy I will generate another 1 min

**Reply by cirospaciari** [reply]
> What that crazy I will generate another 1 min

People just joined today using this link

**Reply by cirospaciari** [reply]
Here is fresh generate for Never expires, no especial permission needed
https://discord.gg/jZCvjFpJrp

**Comment by AmirHmZz** [comment]


**Reply by cirospaciari** [reply]
VSCode

**Comment by AmirHmZz** [comment]
@cirospaciari Do you have any uncommitted changes in `src` directory? I'm I free to open PR to files in `src`?

**Reply by cirospaciari** [reply]
Yes i have the CLI code, some fixes, and some new native code, but you can open an PR to files in src and i do the merges locally if i need.

**Reply by cirospaciari** [reply]
I will recommend you to wait until this weekend, because i will do a lot of commits, but feel free to send any PR i will love to see you adding more and more to this project. Let's make it better and better together :mechanical_arm:

**Comment by cirospaciari** [comment]
Most of this is already done, and the entire socketify.py, uv.py and native.py files will be rewritten in C++ using HPy soon

asgi.py and wsgi.py will be rewritten in another step

This will be locked because of the next steps in the current development stage.
